ibm/resource_ibm_pi_instance.go

```diff
@@ -516,6 +526,27 @@ func resourceIBMPIInstanceCreate(d *schema.ResourceData, meta interface{}) error
 		body.StorageConnection = sc.(string)
 	}
 
+	if lrc, ok := d.GetOk(helpers.PIInstanceLicenseRepositoryCapacity); ok {
+
+		// check if using vtl image
+		imageClient := st.NewIBMPIImageClient(sess, powerinstanceid)
+
+		// check if vtl image is stock image
+		imageData, err := imageClient.GetStockImage(imageid, powerinstanceid)
+		if err != nil {
+			// check if vtl image is cloud instance image
+			imageData, err = imageClient.Get(imageid, powerinstanceid)
+			if err != nil {
+				return fmt.Errorf("Image doesn't exist. %e", err)
+			}
+		}
+
+		if imageData.Specifications.ImageType == "stock-vtl" {
```

Comment:
Is `imageData.Specifications.ImageType` value as "stock-vtl" same for cloud instance image?

Reply:
Yes